apps/opencs/model/doc/document.cpp

```diff
@@ -289,19 +288,23 @@ CSMDoc::Document::Document (const Files::ConfigurationManager& configuration,
 
     if (mNew || !boost::filesystem::exists (mProjectPath))
     {
-        boost::filesystem::path customFiltersPath (configuration.getUserDataPath());
-        customFiltersPath /= "defaultfilters";
-
-        std::ofstream destination (mProjectPath.string().c_str(), std::ios::binary);
-
-        if (boost::filesystem::exists (customFiltersPath))
-        {
-            destination << std::ifstream(customFiltersPath.string().c_str(), std::ios::binary).rdbuf();
-        }
-        else
-        {
-            destination << std::ifstream(std::string(mResDir.string() + "/defaultfilters").c_str(), std::ios::binary).rdbuf();
-        }
+        boost::filesystem::path filtersPath (configuration.getUserDataPath());
+        filtersPath /= "defaultfilters";
+
+        boost::filesystem::ofstream destination(boost::filesystem::path(mProjectPath.string().c_str()), std::ios::out | std::ios::binary);
```

Comment:
Why is it necessary to convert mProjectPath from a boost::filesystem::path to
1) std::string
2) then to C-string
3) back to boost::filesystem::path?

Is it not possible to cut down on conversions here and in mResDir case?